WatchFaceKotlin/wear/src/main/java/com/example/android/wearable/watchfacekotlin/watchface/AnalogWatchFaceRenderer.kt

```diff
@@ -0,0 +1,359 @@
+         */
+        canvas.save()
+
+        canvas.rotate(hoursRotation, centerX, centerY)
+        canvas.drawLine(
+            centerX,
+            centerY - analogWatchFace.centerGapOffsetAndCircleRadius,
+            centerX,
+            centerY - analogWatchFace.hourHand.dimensions.height,
+            analogWatchFace.hourHand.paint
+        )
+
+        canvas.rotate(minutesRotation - hoursRotation, centerX, centerY)
+        canvas.drawLine(
+            centerX,
+            centerY - analogWatchFace.centerGapOffsetAndCircleRadius,
+            centerX,
+            centerY - analogWatchFace.minuteHand.dimensions.height,
+            analogWatchFace.minuteHand.paint
+        )
+
+        /*
+         * Ensure the "seconds" hand is drawn only when we are in interactive mode.
+         * Otherwise, we only update the watch face once a minute.
+         */
+        if (!_ambient) {
+            canvas.rotate(secondsRotation - minutesRotation, centerX, centerY)
+            canvas.drawLine(
+                centerX,
+                centerY - analogWatchFace.centerGapOffsetAndCircleRadius,
+                centerX,
+                centerY - analogWatchFace.secondHand.dimensions.height,
+                analogWatchFace.secondHand.paint
+            )
+        }
+
+        canvas.drawCircle(
+            centerX,
+            centerY,
+            analogWatchFace.centerGapOffsetAndCircleRadius.toFloat(),
+            analogWatchFace.ticks.paint
+        )
+
+        /* Restore the canvas' original orientation. */
+        canvas.restore()
+    }
+
+    fun updateWatchFaceComplication(complicationId: Int, complicationData: ComplicationData) {
+        analogWatchFace.updateWatchFaceComplication(complicationId, complicationData)
+    }
+
+    fun checkTapLocation(tapType: Int, x: Int, y: Int, eventTime: Long) {
+        analogWatchFace.checkTapLocation(tapType, x, y, eventTime)
+    }
+
+    fun setLowBitAndBurnInProtection(properties: Bundle) {
+        val lowBitAmbient =
+            properties.getBoolean(CanvasWatchFaceService.PROPERTY_LOW_BIT_AMBIENT, false)
+        val burnInProtection =
+            properties.getBoolean(CanvasWatchFaceService.PROPERTY_BURN_IN_PROTECTION, false)
+
+        analogWatchFace.setLowBitAndBurnInProtection(lowBitAmbient, burnInProtection)
+    }
+
+    /**
+     * Second hand should only be visible in active mode.
+     */
+    private fun secondHandVisible(): Boolean {
+        return _visible && !_ambient
+    }
+
+    private fun secondHandAnimationActive(): Boolean {
+        return secondHandAnimationJob?.isActive ?: false
+    }
+
+    /**
+     * Dims display in mute mode.
+     */
+    fun toggleDimMode(muteMode: Boolean) {
+        analogWatchFace.setMuteMode(muteMode)
+    }
+
+    /**
+     * Starts the coroutine/flow "timer" to animate the second hand every second when the
+     * watch face is active.
+     */
+    @InternalCoroutinesApi
```

Comment:
Why is this such annotated?

Reply:
@codingjeremy you can remove the annotation. It was needed in a previous version of the Coroutines library, but it's not needed anymore. Don't forget to remove it in all the other spots as well.

Reply:
Good call. I must have had the older version when I initially created this. Removed.